unittests/operators/matrix_op.cpp

```diff
@@ -16,6 +16,9 @@ TEST(OperatorExpressions, checkMatrixOpsUnary) {
   utils::checkEqual((-op).to_matrix({{0, 2}}),
                     -1.0 * utils::position_matrix(2));
   utils::checkEqual(op.to_matrix({{0, 2}}), utils::position_matrix(2));
+  auto adjoint = op.adjoint();
+  utils::checkEqual(adjoint.to_matrix({{0, 2}}),
+                    utils::position_matrix(2).adjoint());
 }
 
 TEST(OperatorExpressions, checkMatrixOpsConstruction) {
```

Comment:
It would be good to add the following tests: 
- for each checkEqual in checkPreBuiltMatrixOps, check that op * op.adjoint() equals the identity
- the same goes for checkCustomMatrixOps

That gives us some very basic coverage of adjoints of sums and products. Additionally, it would be good to add a test each thinking about a hard case to the product_op tests and one to the sum_op tests.